contracts/protocol/facets/Entity.sol

```diff
@@ -740,17 +741,27 @@ contract EntityFacet is Context, EntityErrors, Access, IEntityEvents {
                         if (j != facilitatorsLength - 1)
                             associatedEntities[j] = associatedEntities[facilitatorsLength - 1];
                         associatedEntities.pop();
-
-                        emit AssociatedEntityRemoved(_associatedRole, _sellerId, associatedEntityId);
+                        found = true;
+                        // stack too deep workaround
+                        _emitAssociatedEntityRemoved(_associatedRole, _sellerId, associatedEntityId);
                         break;
                     }
                 }
             }
+            if (!_add && !found) revert NoSuchAssociatedEntity(_associatedRole, _sellerId, associatedEntityId);
```

Comment:
I suppose this could go either:
- outside the for loop (line 754), so it's checked only at the end. In this case `associatedEntityId` can be omitted.
- in line 751 if you still want to keep check per entity id. But then it can be simplified (i.e. checking only `if (!found) ... ` and `bool found;` can be moved to line 739.

Reply:
Makes sense.